### PR TITLE
fix(sql): avoid unexpected errors when over() clause for analytic function is missing

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Also, disallows analytic functions in non-analytic context. That's to match with Postgres behavior.

**Important note**: this is a compatibility breaking change.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 15 / 15 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/analytic/RankFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/analytic/AnalyticContextImpl.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 41 / 48 (85.42%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/analytic/RowNumberFunctionFactory.java|17|23|73.91%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/analytic/RankFunctionFactory.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/analytic/AnalyticContextImpl.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|5|5|100.00%|

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java [hunk 1]
```diff
@@ -57,528 +57,6 @@ public void testAmbiguousFunction() throws Exception {
         );
     }
 
-    @Test
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java:60]: `SqlCodeGeneratorTest` is huge, so it's worth splitting it into specific test suites when possible.

**Review by bluestreak01 — APPROVED** [review]

File: core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java:60]: cool 👍

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 41 / 48 (85.42%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/analytic/RowNumberFunctionFactory.java|17|23|73.91%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/analytic/RankFunctionFactory.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/analytic/AnalyticContextImpl.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|5|5|100.00%|

**Pull request merged**